### PR TITLE
Factor mustache -> modules/lang-mustache

### PR DESCRIPTION
**Pull request by rmuir**
This is the last of the scripting engines to refactor to a module.

It removes a third party dependency from core, removes third party dependency from classpath, and allows us to isolate this thing better, plus it is in my way right now as far as cleanups to core, due to some of the reflection it does (we can submit fixes, but still, lets isolate it).

### CODE REVIEW

File: modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/50_msearch.yaml [hunk 1]
```diff
@@ -51,3 +51,4 @@
           - query:
               { "template": { "query": { "term": { "foo": { "value": "{{template}}" } } }, "params": { "template": "bar" } } }
   - match: { responses.0.hits.total: 1 }
+
```
Comment (jpountz) [inline modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/50_msearch.yaml]: This particular test seems to be more testing msearch than mustache. I think it should stay in the main folder for rest tests minus the mustache part, and then we can have a dedicated mustache test for the msearch integration if we want?

**Comment by jpountz** [comment]
I left one comment about a rest test move but other than that it looks good to me

File: modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/50_msearch.yaml (hunk 1 above)
Comment (rmuir) [inline modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/50_msearch.yaml]: Agree, the test is messy, but so are many tests (see messy tests package). 

If me (the guy simply refactoring stuff) has to fix all of these tests, i'm going to destabilize the build: guaranteed. Thats why I simply move stuff around: the onus cannot be on me.

Alternatively, I am happy to delete the test. But i guess I am saying, I do not think it should be fixed on this issue.

**Comment by rmuir** [comment]
I renamed that rest test to 50_messy_test_msearch.yaml and added it to the messy tests list.

**Comment by jpountz** [comment]
That works for me, thanks, it's more obvious the test is misplaced now. LGTM

**Pull request merged**